Simplified/Reader2/LCP/LCPLibraryService.swift

```diff
@@ -0,0 +1,126 @@
+//
+//  LCPLibraryService.swift
+//  r2-testapp-swift
+//
+//  Created by Mickaël Menu on 01.02.19.
+//
+//  Copyright 2019 Readium Foundation. All rights reserved.
+//  Use of this source code is governed by a BSD-style license which is detailed
+//  in the LICENSE file present in the project repository where this source code is maintained.
+//
+
+#if LCP
+
+import Foundation
+import UIKit
+import R2Shared
+import ReadiumLCP
+
+
+@objc class LCPLibraryService: NSObject, DRMLibraryService {
+
+    private var lcpService =  R2MakeLCPService()
+    
+    /// [LicenseDocument.id: passphrase callback]
+    private var authenticationCallbacks: [String: (String?) -> Void] = [:]
+
+    var brand: DRM.Brand {
+        return .lcp
+    }
+    
+    func canFulfill(_ file: URL) -> Bool {
+        return file.pathExtension.lowercased() == "lcpl"
+    }
+    
+    /// Fulfill LCP license publication
+    /// This function was added for compatibility with Objective-C NYPLMyBooksDownloadCenter.
+    /// - Parameters:
+    ///   - file: LCP license file.
+    ///   - completion: Complition is called after a publication was downloaded or an error received.
+    ///   - localUrl: Downloaded publication URL.
+    ///   - downloadTask: `URLSessionDownloadTask` that downloaded the publication.
+    ///   - error: `NSError` if any.
+    @objc func fulfill(_ file: URL, completion: @escaping (_ localUrl: URL?, _ downloadTask: URLSessionDownloadTask?, _ error: NSError?) -> Void) {
+        lcpService.importPublication(from: file, authentication: self) { result, error in
+            var nsError: NSError?
+            if let error = error {
+                let domain = "SimplyE.LCPLibraryService"
```

Comment:
is this error for the user or for logging? if it's for logging the domain should be something more specific, basically a developer-friendly summary of the error that can help us identify what kind of error it is.

Reply:
This is for logging, the error goes in metadata to NYPLErrorLogger. Its localizedDescription contains more info about the error (that's defined in LCPError in readium-lcp-swift, and contains human readable explanation). Not sure what I can pass in domain here.

Reply:
Here's an example of some summaries on Crashlytics. BTW I think I can give you access if you want. 

![crashlytics-summary-example](https://user-images.githubusercontent.com/289104/98139593-3987a880-1e79-11eb-9877-b70e2f715546.png)

As you can see from those summaries I not only have an idea of where to look in the code, but also what the user is experiencing. This way I can browse Crashlytics starting from the user point of view (for example, "users are experiencing book download errors") which has become increasingly important to diagnose issues quickly. Keep in mind that adding a summary variation, e.g. "\(book.distributor) download error", will create a separate group for every single distributor. Sometimes it's good, sometimes it's too much. There's no rule in these cases, up to our common sense/best judgement.

Also keep in mind that only the summary (i.e. the NSError.domain) and the code (NSError.code) are searchable directly in Crashlytics. So I'm trying to define a way to use them orthogonally: the summary is the more broad type of error, and the codes ideally can traverse different kind of summaries (e.g. the "noURL" can be applied to a lot of different error situations. But currently codes are not defined very well. I'm definitely open to ideas.

Reply:
I see, I can send error messages directly in domain then, e.g.: "LCPLibraryService: License signature does not match."

Reply:
yeah something like that, maybe even more user centric, like in this case it seems like a fulfillment error so we could mention that. The idea is that the summary/domain should help me identify the user problem, and the code the actual error code-level cause. For example signature mismatch sounds more like it should be mapped to a code, because it can happen for LCP, Adobe, or other.... again I'm making this up as we go and if you have ideas I'm interested in hearing them. Identifying errors experienced by users has been difficult.

Reply:
I can add "LCPLibraryService.fulfill: \(errorDescription)", that will directly point to the source or error, file and function, and error description received from LCP service. Adobe DRM reports errors in a different manner, with codes and summaries, so refactoring that the same way will be useful as well.

Reply:
I was even thinking just "\(drmKind) fulfillment error" as summary, but I wonder if having the description in the summary like you said might actually be better. What will happen is that if we have N possible descriptions (even with just one char difference), that will create N separate report groups in Crashlytics. So I would leave file and function out of the summary. Having the "fulfill" string is good though, since it will allow to search for all fulfillment errors with different descriptions.

Reply:
Ok, in this case I'll kepp only "LCP fulfillment error" in summary. We still can save the error message in message then?

Reply:
yeah if you pass in an NSError to NYPLErrorLogger, everything in the error object will be logged.

Reply:
ok changed the format